front-end/js/dynamicLayers.js

```diff
@@ -107,60 +107,64 @@ function highlightLowestSvg(viewId, svg, layerId) {
 
 function renderTiles(viewId, viewportX, viewportY, vpW, vpH, optionalArgs) {
     var gvd = globalVar.views[viewId];
+    var numLayers = gvd.curCanvas.layers.length;
     var viewClass = ".view_" + viewId;
     var tileW = globalVar.tileW;
     var tileH = globalVar.tileH;
 
+    // check # of tile layers
+    if (d3.selectAll(viewClass + ".mainsvg.tiling").size() == 0) return null;
+
     // get tile ids
     var tileIds = getTileArray(viewId, viewportX, viewportY, vpW, vpH);
 
     // assign tile ids to tile
     // and use data joins to remove old tiles and get new tiles
     var tileDataJoins = d3
-        .select(viewClass + ".mainsvg:not(.static)")
+        .select(viewClass + ".mainsvg.tiling")
         .selectAll("svg")
         .data(tileIds, function(d) {
             return d;
         });
 
     if (tileDataJoins.exit().size()) {
         // update gvd.renderData
-        var numLayers = gvd.curCanvas.layers.length;
-        gvd.renderData = [];
-        for (var i = 0; i < numLayers; i++) gvd.renderData.push([]);
-        tileDataJoins.each(function(d) {
-            var tileId = d[0] + " " + d[1] + " " + gvd.curCanvasId;
-            for (var i = 0; i < numLayers; i++)
-                gvd.renderData[i] = gvd.renderData[i].concat(
-                    gvd.tileRenderData[tileId][i]
-                );
-        });
-        for (var i = 0; i < numLayers; i++) {
-            var mp = {};
-            gvd.renderData[i] = gvd.renderData[i].filter(function(d) {
-                return mp.hasOwnProperty(JSON.stringify(d))
-                    ? false
-                    : (mp[JSON.stringify(d)] = true);
-            });
-        }
+        for (var i = 0; i < numLayers; i++)
+            if (gvd.curCanvas.layers[i].fetchingScheme == "tiling") {
+                gvd.renderData[i] = [];
+                tileDataJoins.each(function(d) {
+                    var tileId = d[0] + " " + d[1] + " " + gvd.curCanvasId;
+                    gvd.renderData[i] = gvd.renderData[i].concat(
+                        gvd.tileRenderData[tileId][i]
+                    );
+                });
 
-        // remove exit (invisible) tiles
-        d3.selectAll(viewClass + ".mainsvg:not(.static)").each(function() {
-            var tiles = d3
-                .select(this)
-                .selectAll("svg")
-                .data(tileIds, function(d) {
-                    return d;
+                // deduplicate
+                var mp = {};
+                gvd.renderData[i] = gvd.renderData[i].filter(function(d) {
+                    return mp.hasOwnProperty(JSON.stringify(d))
+                        ? false
+                        : (mp[JSON.stringify(d)] = true);
                 });
-            tiles.exit().remove();
-        });
+
+                // remove exit (invisible) tiles
+                d3.select(".layerg.layer" + i)
```

Comment:
I'm not sure if this is ok if there are multiple views?

Reply:
good point, this should be a bug